app/modals/receive-stx/receive-stx-modal.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FC, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import Qr from 'qrcode.react';
+import { Text, Modal, Button, Flex, Box, useClipboard } from '@blockstack/ui';
+
+import { TxModalHeader, TxModalFooter } from '../transaction/transaction-modal-layout';
+import { homeActions, selectReceiveModalOpen } from '../../store/home/home.reducer';
+
+interface ReceiveStxModalProps {
+  address: string;
+}
+
+export const ReceiveStxModal: FC<ReceiveStxModalProps> = ({ address }) => {
+  const dispatch = useDispatch();
+  const modalOpen = useSelector(selectReceiveModalOpen);
+  const copyAddressToClipboard = useClipboard(address);
+  const [buttonText, setButtonText] = useState('Copy address');
```

Comment:
Small nit: I wonder why we need to use `useState` here? we can just use `{hasCopied, onCopy} = useClipboard(address);` and `hasCopied ? 'Copy address' : 'Copied'`.

Reply:
Agree that's better